pkg/controller/core/dynamicresourceallocationconfig_controller.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/dra"
+	"sigs.k8s.io/kueue/pkg/features"
+)
+
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=dynamicresourceallocationconfigs,verbs=get;list;watch
+
+// DynamicResourceAllocationConfigReconciler keeps the global DRA mapper's DeviceClass->logical-resource
+// mapping in sync with the singleton DynamicResourceAllocationConfig object.
+// It is only registered when the DynamicResourceAllocation feature gate is enabled.
+type DynamicResourceAllocationConfigReconciler struct {
+	client client.Client
+	log    logr.Logger
+}
+
+func NewDRAConfigReconciler(client client.Client) *DynamicResourceAllocationConfigReconciler {
+	return &DynamicResourceAllocationConfigReconciler{
+		client: client,
+		log:    ctrl.Log.WithName("dra-config-reconciler"),
+	}
+}
+
+func (r *DynamicResourceAllocationConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if !features.Enabled(features.DynamicResourceAllocation) {
+		return ctrl.Result{}, nil
+	}
+
+	var cfg kueuealpha.DynamicResourceAllocationConfig
+	if err := r.client.Get(ctx, req.NamespacedName, &cfg); err != nil {
+		// TODO: this should never happen. But if a user deletes the singleton DRA by mistake, we should
+		//  have a way to gracefully handle it. May be adding a finalizer to the singleton DRAConfig object?
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.log.V(2).Info("Reconciling DynamicResourceAllocationConfig", "name", req.NamespacedName)
```

Comment:
```suggestion

	log := ctrl.LoggerFrom(ctx)
	log.V(2).Info("Reconcile DynamicResourceAllocationConfig")
```

In the reconciliation, we should use the propagated logger since the propagated one has reconciled request for something else and other information by default.